src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -511,6 +511,28 @@ QList<int> QgsProcessingParameters::parameterAsEnums( const QgsProcessingParamet
   return result;
 }
 
+QString QgsProcessingParameters::parameterAsEnumString( const QgsProcessingParameterDefinition *definition, const QVariantMap &parameters, const QgsProcessingContext &context )
+{
+  if ( !definition )
+    return QString();
+
+  return parameterAsEnumString( definition, parameters.value( definition->name() ), context );
+}
+
+QString QgsProcessingParameters::parameterAsEnumString( const QgsProcessingParameterDefinition *definition, const QVariant &value, const QgsProcessingContext &context )
+{
+  if ( !definition )
+    return QString();
+
+  int index = parameterAsInt( definition, value, context );
```

Comment:
I think there's actually two different requirements we've got at play here:

1. A method which converts an integer parameter value back to a string (which you've done here).
and 
2. A method which allows the parameter value to be specified as a string value, but restricting it to a list of available options

For the dxf encoding option we actually need number 2 -- a way of having the encoding option be specified as a string value, so that it's not an integer at all and is not going to change if a particular platform has a different set of encodings available (or a different ordering of encodings).

So the API included here is useful, but it's not a solution to the DXF encoding type issue, which would need to be handled differently.